R/assertthat_checks.R

```diff
@@ -0,0 +1,343 @@
+  )
+}
+
+#' Check Validity of the Minute/Second Portion of the Time Input
+#'
+#' Minutes and seconds are expected to range from 0 to 59
+#'
+#' @param arg The argument to check
+#'
+#' @author Samia Kabi
+#'
+#' @return `TRUE` if the argument is a valid min/sec input, `FALSE` otherwise
+#'
+#' @keywords check
+#'
+#' @noRd
+#'
+#' @examples
+#' assertthat::assert_that(admiral:::is_valid_sec_min(59))
+is_valid_sec_min <- function(arg) {
+  arg %in% 0:59
+}
+on_failure(is_valid_sec_min) <- function(call, env) {
+  paste0(
+    "Argument ",
+    deparse(call$arg),
+    " = ",
+    eval(call$arg, envir = env),
+    " is not a valid min/sec.\n",
+    "Values must be between between 0-59"
+  )
+}
+
+#' Check Validity of the Hour Portion in the Time Input
+#'
+#' Hours are expected to range from 0 to 23
+#'
+#' @param arg The argument to check
+#'
+#' @author Samia Kabi
+#'
+#' @return `TRUE` if the argument is a valid hour input, `FALSE` otherwise
+#'
+#' @keywords check
+#'
+#' @noRd
+#'
+#' @examples
+#' assertthat::assert_that(admiral:::is_valid_hour(20))
+is_valid_hour <- function(arg) {
+  arg %in% 0:23
+}
+on_failure(is_valid_hour) <- function(call, env) {
+  paste0(
+    "Argument ",
+    deparse(call$arg),
+    "=",
+    eval(call$arg, envir = env),
+    " is not a valid hour.\n",
+    "Values must be between 0-23"
+  )
+}
+
+#' Check Validity of the Day Portion in the Date Input
+#'
+#' Days are expected to range from 1 to 31
+#'
+#' @param arg The argument to check
+#'
+#' @author Samia Kabi
+#'
+#' @return `TRUE` if the argument is a day input, `FALSE` otherwise
+#'
+#' @keywords check
+#'
+#' @noRd
+#'
+#' @examples
+#' assertthat::assert_that(admiral:::is_valid_day(20))
+is_valid_day <- function(arg) {
+  arg %in% 1:31
+}
+on_failure(is_valid_day) <- function(call, env) {
+  paste0(
+    "Argument ",
+    deparse(call$arg),
+    " = ",
+    eval(call$arg, envir = env),
+    " is not a valid day.\n",
+    "Values must be between 1-31"
+  )
+}
+
+#' Check Validity of the Month Portion in the Date Input
+#'
+#' Days are expected to range from 1 to 12
+#'
+#' @param arg The argument to check
+#'
+#' @author Samia Kabi
+#'
+#' @return `TRUE` if the argument is a month input, `FALSE` otherwise
+#'
+#' @keywords check
+#'
+#' @noRd
+#'
+#' @examples
+#' assertthat::assert_that(admiral:::is_valid_month(12))
+is_valid_month <- function(arg) {
+  arg %in% 1:12
+}
+on_failure(is_valid_month) <- function(call, env) {
+  paste0(
+    "Argument ",
+    deparse(call$arg),
+    " = ",
+    eval(call$arg, envir = env),
+    " is not a valid month.\n",
+    "Values for month must be between 1-12. ",
+    "Please check the date_imputation input: it should be sepcified as 'dd-mm'"
+  )
+}
+
+is_vars <- function(arg) {
+  inherits(arg, "quosures") && all(map_lgl(arg, quo_is_symbol))
+}
+on_failure(is_vars) <- function(call, env) {
+  paste0(
+    "Argument `",
+    deparse(call$arg),
+    "` is not a list of variables created using `vars()`"
+  )
+}
```

Comment:
This function is called in `extract_duplicate_records()` and `signal_duplicate_records()` only.

Should we replace the calls by `assert_vars()` and remove `is_vars()` from admiral?

Reply:
Removed.